### PR TITLE
A little tweak in amethyst_assets public interface

### PR DESCRIPTION
**Pull request by omni-viral**
Make `Asset` type appear in public methods parameters instead of its `Context`

### CODE REVIEW

File: amethyst_assets/src/asset.rs [hunk 1]
```diff
@@ -23,6 +23,15 @@ pub trait Asset
 where
     Self: Clone + Sized,
 {
+    /// The `Context` type that can produce this asset
+    type Context: Context<Asset=Self, Data=Self::Data, Error=Self::Error>;
+
+    /// The `Data` type the asset can be created from.
+    type Data;
```
Comment (torkleyy) [inline amethyst_assets/src/asset.rs]: If we want these types here, could we remove them from `Context`?

Comment (omni-viral) [inline amethyst_assets/src/asset.rs]: They are just like aliases to write `A::Error` instead of `<A::Context as Context::Error`.

Comment (omni-viral) [inline amethyst_assets/src/asset.rs]: Same as `Item` in `std::IntoIterator`

Comment (torkleyy) [inline amethyst_assets/src/asset.rs]: Okay, fair.

**Comment by torkleyy** [comment]
This is definitely an improvement and makes type inference better. Seems there are no other concerns, so I'm going to merge this PR soon.

**Pull request merged**